samcli/lib/intrinsic_resolver/intrinsic_property_resolver.py

```diff
@@ -270,9 +270,8 @@ def resolve_resources_assets(self, ignore_errors):
                                 asset.source_local_image, ignore_errors, parent_function=asset.source_property
                             )
                         else:
-                            asset.source_path = self.intrinsic_property_resolver(
-                                asset.source_path, ignore_errors, parent_function=asset.source_property
-                            )
+                            image_tag = asset.image_tag
+                            asset.source_local_image = f"{logical_id}:{image_tag}"
```

Comment:
what was the issue here or why did this change?

Reply:
I ran into a bug when testing image assets with `local invoke`. sam-cli was trying to initialize the container using the `asset.source_path` as the image name which resulted in the container failing to start.